### PR TITLE
Add flatmap()

### PR DESCRIPTION
**Pull request by lionel-**
And allow `.type` = "numeric" in `can_simplify()`.

### CODE REVIEW

File: R/coercion.R [hunk 1]
```diff
@@ -55,21 +55,26 @@ can_simplify <- function(x, type = NULL) {
...
-    type <- typeof(type)
   }
 
-  actual <- typeof(x[[1]])
-  safe_int_coercions <- c("integer", "double", "numeric")
-  if (actual == "integer" && type %in% safe_int_coercions) {
-    TRUE
-  } else {
-    actual == type
+  if (actual == "integer" && type %in% c("integer", "double", "numeric")) {
```
Comment (hadley) [inline R/coercion.R]: When does `typeof()` return `numeric`?

File: R/map-vector.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#' \dontrun{
+#' c(mtcars) %>% flatmap(var_select, .type = "character")
+#' }
+#' c(mtcars) %>% flatmap(var_select, .type = "numeric")
+flatmap <- function(.x, .f, ..., .type) {
+  if (!missing(.type)) {
+    if (!is_scalar_character(.type)) {
+      stop(".type must be a string")
+    }
+    if (!can_simplify(.x, .type)) {
```
Comment (hadley) [inline R/map-vector.R]: Shouldn't that be called on the results of `map(.x, .f)`?

Comment (lionel-) [inline R/map-vector.R]: oh yes, sorry

File: R/coercion.R (hunk 1 above)
Comment (lionel-) [inline R/coercion.R]: it's to allow `.type = "numeric"` in `flatmap()` and `as_vector()`.

Comment (hadley) [inline R/coercion.R]: Oooo, I see.

File: R/map-vector.R (hunk 2 above)
Comment (lionel-) [inline R/map-vector.R]: fixed, and the test now catches this.

File: R/coercion.R (hunk 1 above)
Comment (hadley) [inline R/coercion.R]: Could you add a comment to that effect?

Comment (lionel-) [inline R/coercion.R]: here you go

**Comment by hadley** [comment]
Thanks - will submit to CRAN very soon!

**Comment by lionel-** [comment]
cool :)

**Pull request merged**